src/zc/zodbdgc/__init__.py

```diff
@@ -441,6 +468,11 @@ def _insert_ref(references, rname, roid, name, oid):
     by_oid = references.get(name)
     if not by_oid:
         by_oid = references[name] = BTrees.LOBTree.BTree()
+        # XXX: Setting _p_changed is needed when using the pure-python
+        # BTree package (e.g., under PyPy). This is probably a bug in
+        # BTree (LOBTreePy)? Without it, the References example in README.test
+        # fails.
+        references._p_changed = True
```

Comment:
Oh my.  Do you have time to file a [BTrees bug](https://github.com/zopefoundation/BTrees/issues)?